source/fundamentals/class-mapping.txt

```diff
@@ -0,0 +1,326 @@
+using attributes on the property. For more information on ways to customize
+properties in your {+language+} class, see :ref:`csharp-custom-serialization`.
+
+Support Extra Elements
+~~~~~~~~~~~~~~~~~~~~~~
+
+You can design your {+language+} class to store any extra elements in your
+document that don't have matching class properties. To do this your class must
+have a ``BsonDocument`` type property to hold the extra elements.
+
+The following code uses the ``BsonExtraElements`` attribute with the
+``ExtraElements`` property to direct the driver to store extra elements:
+
+.. code-block:: csharp
+
+   public class Person
+   {
+       public string Name { get; set; 
+       public int Age { get; set; }
+       public List<string> Hobbies {get; set;}
+       [BsonExtraElements]
+       public BsonDocument ExtraElements {get; set;}
+   }
+
+You can also support extra elements when initializing a class map as follows:
+
+.. code-block:: csharp
+   
+   BsonClassMap.RegisterClassMap<Person>(classmap => 
+   {
+       classmap.AutoMap();
+       classmap.MapExtraElementsMember(p => p.ExtraElements);
+   });
+
+.. note::
+
+   After the driver serializes a class with extra elements back to BSON, the 
+   extra elements may not be in the same order as they were in the original 
+   document.
+
+Dynamically Serialize Properties
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can use a method to determine whether or not to serialize a property. For
+the driver to automatically use the method when serializing, the method must be
+named ``ShouldSerialize<property name>``. When the driver sees a method with
+this naming convention, it will use the method to determine whether or not to
+serialize properties of the provided property name.
+
+The following example creates a method that only serializes the ``Age`` property
+if its value is greater than ``0``. The driver ignores any properties whose
+values don't meet this requirement:
+
+.. code-block:: csharp
+
+   public class Person 
+   {
+       public string Name { get; set; }
+       public int Age { get; set; }
+       public List<string> Hobbies {get; set;}
+
+       public bool ShouldSerializeAge()
+       {
+          return Age > 0;
+       }
+   }
+
+You can also specify the method while registering a class map:
+
+.. code-block:: csharp
+
+   BsonClassMap.RegisterClassMap<Employee>(classmap => 
+   {
+       classmap.AutoMap();
+       classmap.MapMember(p => c.Age).SetShouldSerializeMethod(
+           obj => ((Person) obj).Age > 0
+       );
+   });
+
+Additional Information
+----------------------
+
+For more information on using {+language+} classes, see :ref:`csharp-poco`.
+
+API Documentation
+-----------------
+
+- `BsonClassMap
+  <{+api-root+}/T_MongoDB_Bson_Serialization_BsonClassMap_1.htm>`__
+- `RegisterClassMap
+  <{+api-root+}/M_MongoDB_Bson_Serialization_BsonClassMap_RegisterClassMap__1_1.htm>`__
+- `AutoMap
+  <{+api-root+}/M_MongoDB_Bson_Serialization_BsonClassMap_AutoMap.htm>`__
```

Comment:
There are more methods mentioned in the document that could be listed here. Just a suggestion to include them.